cloudinit/dmi.py

```diff
@@ -159,9 +198,12 @@ def read_dmi_data(key: str) -> Optional[str]:
     if is_container():
         return None
 
-    if is_FreeBSD():
+    if is_FreeBSD() or is_DragonFlyBSD():
```

Comment:
A key benefit to doing that is that we can fall-back to much, much easier.
So we can use the fast method first, and if that fails go to dmidecode (which then may throw an exception because it's not installed, or in OpenBSD's case, because it's not enabled)